File: src/main/java/com/github/krystianmuchla/skyr/note/sync/NotesSyncResult.java
package com.github.krystianmuchla.skyr.note.sync;

import com.github.krystianmuchla.skyr.note.Note;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record NotesSyncResult(UUID syncId, Instant syncTime, List<Note> modifiedNotes) {
}

File: src/main/kotlin/com/example/skyr/note/sync/NotesSyncResult.kt
package com.example.skyr.note.sync

import com.example.skyr.note.Note
import java.time.Instant
import java.util.*

class NotesSyncResult(val syncId: UUID, val syncTime: Instant, val modifiedNotes: List<Note>? = null)

fun notesSyncResult(notesSync: NotesSync) = NotesSyncResult(notesSync.syncId, notesSync.syncTime)

fun notesSyncResult(notesSync: NotesSync, modifiedNotes: List<Note>) =
    NotesSyncResult(notesSync.syncId, notesSync.syncTime, modifiedNotes)
